crates/sdk/src/masp/shielded_sync/utils.rs

```diff
@@ -0,0 +1,1291 @@
+        &self,
+    ) -> Result<(BlockHeight, xorf::BinaryFuse16), Error> {
+        use serde::Deserialize;
+
+        #[derive(Deserialize)]
+        struct Response {
+            block_height: u64,
+            index: xorf::BinaryFuse16,
+        }
+
+        let _permit = self.shared.semaphore.acquire().await.unwrap();
+
+        let response = self
+            .client
+            .get(self.endpoint("/block-index"))
+            .keep_alive()
+            .send()
+            .await
+            .map_err(|err| {
+                Error::Other(format!(
+                    "Failed to fetch latest masp tx block index: {err}"
+                ))
+            })?;
+        if !response.status().is_success() {
+            let err = Self::get_server_error(response).await?;
+            return Err(Error::Other(format!(
+                "Failed to fetch latest masp tx block index: {err}"
+            )));
+        }
+        let payload: Response = response.json().await.map_err(|err| {
+            Error::Other(format!(
+                "Could not deserialize latest masp tx block index JSON \
+                 response: {err}"
+            ))
+        })?;
+
+        Ok((BlockHeight(payload.block_height), payload.index))
+    }
+}
+
+#[cfg(not(target_family = "wasm"))]
+impl MaspClient for IndexerMaspClient {
+    async fn last_block_height(&self) -> Result<Option<BlockHeight>, Error> {
+        use serde::Deserialize;
+
+        #[derive(Deserialize)]
+        struct Response {
+            block_height: u64,
+        }
+
+        let _permit = self.shared.semaphore.acquire().await.unwrap();
+
+        let response = self
+            .client
+            .get(self.endpoint("/height"))
+            .keep_alive()
+            .send()
+            .await
+            .map_err(|err| {
+                Error::Other(format!(
+                    "Failed to fetch latest block height: {err}"
+                ))
+            })?;
+        if !response.status().is_success() {
+            let err = Self::get_server_error(response).await?;
+            return Err(Error::Other(format!(
+                "Failed to fetch last block height: {err}"
+            )));
+        }
+        let payload: Response = response.json().await.map_err(|err| {
+            Error::Other(format!(
+                "Could not deserialize latest block height JSON response: \
+                 {err}"
+            ))
+        })?;
+
+        Ok(if payload.block_height != 0 {
+            Some(BlockHeight(payload.block_height))
+        } else {
+            None
+        })
+    }
+
+    async fn fetch_shielded_transfers(
+        &self,
+        BlockHeight(mut from): BlockHeight,
+        BlockHeight(to): BlockHeight,
+    ) -> Result<Vec<IndexedNoteEntry>, Error> {
+        use std::ops::ControlFlow;
+
+        use serde::Deserialize;
+        use xorf::Filter;
+
+        #[derive(Deserialize)]
+        struct TransactionSlot {
+            // masp_tx_index: u64,
```

Comment:
Is this going to be used later?

Reply:
the idea was to leverage this, but we ended up not needing that `masp_tx_index` field. we can def remove that commented out field to avoid any confusion